packages/react-native/ReactCommon/react/renderer/imagemanager/platform/ios/react/renderer/imagemanager/RCTImagePrimitivesConversions.h

```diff
@@ -102,13 +122,30 @@ inline static NSURLRequest *NSURLRequestFromImageSource(const facebook::react::I
 
   NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:url];
 
-  /*
-  // TODO(shergin): To be implemented.
-  request.HTTPBody = ...;
-  request.HTTPMethod = ...;
-  request.cachePolicy = ...;
-  request.allHTTPHeaderFields = ...;
-  */
+  NSString *method = @"GET";
+  if (!imageSource.method.empty()) {
+    method = [[NSString alloc] initWithBytesNoCopy:(void *)imageSource.method.c_str()
+                                            length:imageSource.method.size()
+                                          encoding:NSUTF8StringEncoding
+                                      freeWhenDone:NO]
+                 .uppercaseString
+        ?: @"GET";
+  }
+  NSData *body = nil;
+  if (!imageSource.body.empty()) {
+    NSString *bodyString = [[NSString alloc] initWithBytesNoCopy:(void *)imageSource.body.c_str()
+                                                          length:imageSource.body.size()
+                                                        encoding:NSUTF8StringEncoding
+                                                    freeWhenDone:NO];
+    body = [RCTConvert NSData:bodyString];
```

Comment:
can't we just initialize it with some `NSData` initializer?

Reply:
Seems no suitable initializer...

Reply:
the initializer should be:
```objc
[NSData dataWithBytes:imageSource.body.c_str() length:imageSource.body.size()]
```
We use this also in other locations, like [here](https://github.com/facebook/react-native/blob/main/packages/react-native/React/CxxBridge/RCTObjcExecutor.mm#L77)

Reply:
@cipolleschi I updated. But they are maybe some different when bytes is not a valid UTF-8 bytes?